tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs

```diff
@@ -44,25 +49,100 @@ public void DecodeBmp<TPixel>(TestImageProvider<TPixel> provider)
         }
 
         [Theory]
-        [WithFile(F, CommonNonDefaultPixelTypes)]
-        public void BmpDecoder_IsNotBoundToSinglePixelType<TPixel>(TestImageProvider<TPixel> provider)
+        [WithFileCollection(nameof(BitfieldsBmpFiles), PixelTypes.Rgba32)]
+        public void BmpDecoder_CanDecodeBitfields<TPixel>(TestImageProvider<TPixel> provider)
             where TPixel : struct, IPixel<TPixel>
         {
             using (Image<TPixel> image = provider.GetImage(new BmpDecoder()))
             {
-                image.DebugSave(provider, "bmp");
+                image.DebugSave(provider);
                 image.CompareToOriginal(provider);
             }
         }
 
+        [Theory]
+        [WithFile(Rgba321010102, PixelTypes.Rgba32)]
+        public void BmpDecoder_CanDecodeBitfields_WithUnusualBitmasks<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage(new BmpDecoder()))
+            {
+                image.DebugSave(provider);
+
+                // Choosing large tolerance of 6.1 here, because for some reason with the MagickReferenceDecoder the alpha channel is always 191 instead of 255.
```

Comment:
Youthink there's a bug in the reference decoder then?

Reply:
I have loaded this image with GIMP, exported it as PNG and compared to my results. It produces the same results as i have, including the alpha channel, which gives me more confidence, that it is correct. I could add this to the ReferenceOutput and compare against it, if you like.

Reply:
No, don't bother, just add to the comment saying you have done this.